server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -352,7 +358,15 @@ public final DocumentParserContext addIgnoredFieldFromContext(IgnoredSourceField
             } else {
                 assert ignoredFieldWithNoSource != null;
                 assert ignoredFieldWithNoSource.value() == null;
+
+                // Don't expand dots when we use the parser to get part of the source
+                // since dot expanding can modify a document in a way that makes it invalid.
+                // See #encodeFlattenedToken().
+                boolean old = path().isWithinLeafObject();
+                path().setWithinLeafObject(true);
                 Tuple<DocumentParserContext, XContentBuilder> tuple = XContentDataHelper.cloneSubContext(this);
```

Comment:
Would just unwrapping the parser in `cloneSubContextParserConfiguration(...)` be possible and if so easier?

Reply:
Yes, i think it's cleaner.

Reply:
I tried using your approach with `getWrappedParser()` but `DotExpandingXContentParser` can wrap more than one parser and i am not sure what's going on at that time.